Messenger/src/test/java/com/epam/testmodule/FileModeTest.java

```diff
@@ -0,0 +1,134 @@
+        Assertions.assertEquals(expected, extracted);
+
+    }
+
+    @Test
+    public void StartEngine_WhenCorrectString_ShouldSpiltStringAndSendMessage() {
+        //given
+        String parseFile = "Helloween`31 October`mike@gmail.com";
+        String[] actualSplitFiles = parseFile.split("`");
+        String[] expectedSplitFiles = {"Helloween", "31 October", "mike@gmail.com"};
+        Template template = new Template();
+        Client client = new Client();
+        template.setEventName(actualSplitFiles[0]);
+        template.setDate(actualSplitFiles[1]);
+        client.setMail(actualSplitFiles[2]);
+        FileMode fileMode = spy(new FileMode());
+        MailServer mailServer = mock(MailServer.class);
+        TemplateEngine templateEngine = mock(TemplateEngine.class);
+        Messenger messengerSpy = Mockito.spy(new Messenger(mailServer, templateEngine));
+        //when
+        fileMode.startEngine(parseFile);
+        messengerSpy.sendMessage(client, template);
+        //assert
+        assertArrayEquals(actualSplitFiles, expectedSplitFiles);
+        verify(fileMode).startEngine(parseFile);
+        verify(messengerSpy, Mockito.times(1)).sendMessage(client, template);
+        verify(messengerSpy).sendMessage(client, template);
+    }
+    @Test
+    public void getInfoFromUser_ShouldReturnAnyString(){
+        //given
+        FileMode fileMode = new FileMode();
+        String input = "E:\\TempDir\\data2.txt";
+        InputStream in = new ByteArrayInputStream(input.getBytes());
+        System.setIn(in);
+        //assert
+        assertEquals("E:\\TempDir\\data2.txt",fileMode.getInfoFromUser());
+    }
+    @Test
+    public void start_WhenUserInputCorrectPathToFile_ShouldSendMessage() throws IOException {
+        //given
+        FileMode fileMode = spy(new FileMode());
+        FileMode fileMode1 = new FileMode();
+        doReturn("E:\\TempDir\\data2.txt").when(fileMode).getInfoFromUser();
+        doReturn(true).when(fileMode).checkFilePath("E:\\TempDir\\data2.txt");
+        doReturn("Event = Helloween Data = 31 October Client mail = mike@gmail.com").when(fileMode).getDataFromFile("E:\\TempDir\\data2.txt");
+        String actualDataFromFile = fileMode1.getDataFromFile("E:\\TempDir\\data2.txt");
+        String expectedDataFromFile = "Event = Helloween Data = 31 October Client mail = mike@gmail.com";
+        //when
+        fileMode.start();
+        //assert
+        Assertions.assertEquals(actualDataFromFile,expectedDataFromFile);
+        verify(fileMode,times(1)).start();
+        verify(fileMode,times(1)).checkFilePath("E:\\TempDir\\data2.txt");
+        verify(fileMode,times(1)).getInfoFromUser();
+        verify(fileMode,times(1)).getDataFromFile("E:\\TempDir\\data2.txt");
+        verify(fileMode,times(1)).parseFile("Event = Helloween Data = 31 October Client mail = mike@gmail.com");
+        verify(fileMode,times(1)).startEngine("Helloween`31 October`mike@gmail.com");
+    }
+
```

Comment:
seeems like the test for the alternative processing flow of 'start' method is missing